app/assets/v2/js/pages/profile-trust.js

```diff
@@ -657,6 +657,152 @@ Vue.component('sms-verify-modal', {
   }
 });
 
+Vue.component('ens-verify-modal', {
+  data: function() {
+    return {
+      showValidation: false,
+      static_url: document.contxt.STATIC_URL,
+      validationStep: 0,
+      validationError: '',
+      validationErrorMsg: '',
+      validated: false,
+      verificationEthAddress: '',
+    };
+  },
+  mounted: function() {
+    $(document).on('click', '#verify-ens-link', function(event) {
+      event.preventDefault();
+      this.showValidation = true;
+    }.bind(this));
+    this.checkENSValidation();
+  },
+  template: `
+    <b-modal id="ens-modal" @hide="dismissVerification()" :visible="showValidation" center hide-header hide-footer>
+      <template v-slot:default="{ hide }">
+        <div class="mx-5 mt-5 mb-4 text-center">
+          <div class="mb-3">
+            <h1 class="font-bigger-4 font-weight-bold">Verify with ENS</h1>
+            <p>
+              ENS is a name service built on Ethereum. It offers a secure and decentralized way to address resources using human-readable names.
+              <p class="mb-4">
+                <a href="https://ens.domains/about">Learn more.</a>
+              </p>
+            </p>
+          <div>
+            <h2 class="font-bigger-4 font-weight-bold">You need to have these requirements:</h2>
+            <ol>
+              <li>Setup your address. <span :class="\`mr-2 fas fa-\${ validationStep > 1 ? 'check check gc-text-green' : 'times gc-text-pink' }\`"></span>
+                <input v-model="verificationEthAddress" class="form-control" type="text" @onchange="testVerification()">
+                <a href="#" @click.prevent.stop="pullEthAddress()">Pull address from Metamask</a><br>
+              </li>
+              <li>Your address has an ENS domain associated, you can get one <a href="https://app.ens.domains/">here</a>. <span :class="\`mr-2 fas fa-\${ validationStep > 2 ? 'check gc-text-green' : 'times gc-text-pink' }\`"></span></li>
```

Comment:
Can you display the ens name if reverse record is set at this stage?

Reply:
i added it